codec-dns/src/main/java/io/netty/handler/codec/dns/DefaultDnsRecordDecoder.java

```diff
@@ -106,7 +106,19 @@ protected DnsRecord decodeRecord(
      * @param in the byte buffer containing the DNS packet
      * @return the domain name for an entry
      */
-    protected String decodeName(ByteBuf in) {
+    protected String decodeName0(ByteBuf in) {
```

Comment:
Why need to create this `decodeName0` method? Is it for providing a hook method to allow people to change the behavior?